core/src/main/java/apoc/trigger/TriggerHandler.java

```diff
@@ -57,15 +58,17 @@ private enum Phase {before, after, rollback, afterAsync}
 
     public static final String TRIGGER_REFRESH = "apoc.trigger.refresh";
 
-    private final ConcurrentHashMap<String, Map<String,Object>> activeTriggers = new ConcurrentHashMap();
+    // Snapshot of installed triggers. The containing map is immutable.
+    private final AtomicReference<Map<String, Map<String, Object>>> triggersSnapshot = new AtomicReference<>(Map.of());
+
     private final Log log;
     private final GraphDatabaseService db;
     private final DatabaseManagementService databaseManagementService;
     private final ApocConfig apocConfig;
     private final Pools pools;
     private final JobScheduler jobScheduler;
 
-    private long lastUpdate;
+    private volatile long lastUpdate;
```

Comment:
Why does this need to be thread safe?

Reply:
It can be set and read from multiple different threads at the same time (from scheduled thread and from procedures calling into here).